### PR TITLE
remove sync all

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

assign @howardjohn @ramaraochavali @stevenctl

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
was this only added due to the handler syncing thing? Not to explicitly run them all "In Order"?

Hopefully not the latter since that means we have other bugs; shouldn't depend on ordering.

I think this is probably the right path but want Steven and Rama to look as well

**Comment by hzxuzhonghu** [comment]
>was this only added due to the handler syncing thing?

Yes @howardjohn because of the recent client-go upgrade

**Comment by hzxuzhonghu** [comment]
suspect the sync order has some dependency. The tests failed are all cross network calling. Maybe the network update should trigger some additional sync cb.

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -779,19 +714,14 @@ func (c *Controller) Run(stop <-chan struct{}) {
 	st := time.Now()
 	if c.opts.NetworksWatcher != nil {
 		c.opts.NetworksWatcher.AddNetworksHandler(c.reloadNetworkLookup)
-		c.reloadMeshNetworks()
-		c.reloadNetworkGateways()
+		c.reloadNetworkLookup()
 	}
+	c.onDefaultNetworkChange()
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:719]: on network change, we have to refresh the network. There is other way though.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**